osu.Game.Rulesets.Osu/OsuDifficulty/Skills/Skill.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.OsuDifficulty.Preprocessing;
+using osu.Game.Rulesets.Osu.OsuDifficulty.Utils;
+
+namespace osu.Game.Rulesets.Osu.OsuDifficulty.Skills
+{
+    public abstract class Skill
+    {
+        /// <summary>
+        /// Strain values are multiplied by this number for the given skill. Used to balance the value of different skills between each other.
+        /// </summary>
+        protected abstract double SkillMultiplier { get; }
+
+        /// <summary>
+        /// Determines how quickly strain decays for the given skill.
+        /// For example a value of 0.15 indicates that strain decays to 15% of it's original value in one second.
+        /// </summary>
+        protected abstract double StrainDecayBase { get; }
+
+        /// <summary>
+        /// The note that will be processed.
+        /// </summary>
+        protected OsuDifficultyHitObject Current;

```

Comment:
This member variable should be a thing. Just let `StrainValue` take a `OsuDifficultyHitObject` as its argument.